web/src/engine/keyboard/src/kmxPlus/layr.ts

```diff
@@ -0,0 +1,105 @@
+import { KeySpec, Keys } from './keys.js';
+import { Strs } from './strs.js';
+import { decodeNumber } from './utils.js';
+
+export type Layer = {
+  id: string;
+  modifiers: number;
+  keys: KeySpec[][];
+};
+
+export type LayerList = {
+  layers: Layer[],
+  minDeviceWidth: number,
+  name: string
+};
+
+export class Layr {
+  readonly name = 'layr';
+  readonly data: Uint8Array;
+  readonly layerLists: Readonly<LayerList>[]
+
+  constructor(rawData: Uint8Array, strs: Strs, keys: Keys) {
+    this.data = rawData;
+
+    const layerListCount = decodeNumber(rawData, 8);
+
+    const layerListTableOffset = 24;
+    const LAYER_LIST_ENTRY_SIZE = 16;
+    const layerTableOffset = layerListTableOffset + LAYER_LIST_ENTRY_SIZE * layerListCount;
+
+    const layers = this.processLayers(layerTableOffset, strs, keys);
+
+    const layerLists: LayerList[] = [];
+
+    for(let i = 0; i < layerListCount; i++) {
+      const rowOffset = layerListTableOffset + i * LAYER_LIST_ENTRY_SIZE;
+
+      const name = strs.entries[decodeNumber(rawData, rowOffset)];
+      const firstLayerIndex = decodeNumber(rawData, rowOffset + 4);
+      const layerCount = decodeNumber(rawData, rowOffset + 8);
+      const minDeviceWidth = decodeNumber(rawData, rowOffset + 12);
+
+      const listLayers: Layer[] = [];
+      for(let j = 0; j < layerCount; j++) {
+        listLayers.push(layers[firstLayerIndex + j]);
+      }
+
+      layerLists.push({
+        name, minDeviceWidth, layers: listLayers
+      });
+    }
+
+    this.layerLists = layerLists;
+  }
+
+  private processLayers(layerTableOffset: number, strs: Strs, keys: Keys) {
```

Comment:
This method may include some "heavy lifting" on behalf of the `convertLayerList` method.